### PR TITLE
Completed feature (#4), user can now see selected product detail on t…

### PR DESCRIPTION
**Pull request by YunCYang**
…he frontend

wicked_sale_feature_4

### CODE REVIEW

**Comment by YunCYang** [comment]
image

I encountered an error that says I am updating state on an unmounted component. I am not too sure how to solve this issue. All the functions work properly even if I ignore the bug.

File: client/components/app.jsx [hunk 1]
```diff
@@ -19,13 +25,32 @@ export default class App extends React.Component {
       .finally(() => this.setState({ isLoading: false }));
   }
 
+  setView(name, params) {
+    this.setState({
+      message: this.state.message,
+      isLoading: this.state.isLoading,
```
Comment (thebearingedge) [inline client/components/app.jsx]: You can remove `message` and `isLoading` from all of your `App` state. It was just for testing purposes.

File: client/components/product-detail.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+const ProductDetails = props => {
+  const [product, setProduct] = React.useState(null);
+
+  React.useEffect(
+    () => {
+      fetch(`/api/products/${props.id}`)
+        .then(res => res.json())
+        .then(res => setProduct(res));
+    }, [product]
```
Comment (thebearingedge) [inline client/components/product-detail.jsx]: Are you sure you need this `[product]` or would an empty array work fine? `[]`

File: client/components/product-list-item.jsx [hunk 3]
```diff
@@ -11,7 +13,7 @@ const ProductListItem = props => {
           <span>{props.product.name}</span>
         </div>
         <div className="product-price-container">
-          <span>{props.product.price}</span>
+          <span>$ {props.product.price}</span>
```
Comment (thebearingedge) [inline client/components/product-list-item.jsx]: The prices are in pennies, so they'll need to be converted to dollars.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by YunCYang** [comment]
wicked_sale_feature_4_fixed_bug

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Please attempt to match the layout in the Figma and example.

**Comment by YunCYang** [comment]
image

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**